src/dinit.cc

```diff
@@ -453,6 +471,90 @@ static int process_commandline_arg(char **argv, int argc, int &i, options &opts)
     return 0;
 }
 
+#if SUPPORT_SELINUX
+// Load the system SELinux policy and transition ourselves to it. When successful,
+// this will cause SELinux labels as per the policy to be attached to processes (and
+// file descriptors owned by those processes). The SELinux framework will begin to
+// enforce restrictions on access based on these labels and the loaded policy.
+//
+// We might lose access to any file descriptors we have open when this is called (since
+// they will still be labelled with the kernel context), so it is best done early (i.e.
+// before we start opening file descriptors).
+//
+// Parameters:
+//   exe - the path that we are invoked with (to calculate our new security
+//         context to tranition into.)
+//
+// Returns:
+//   If we fail to load the system SELinux policy, return false, otherwise,
+//   return true.
+static bool selinux_transition(const char *exe) {
+    // Let's use std::cerr instead of the log for logging messages here.
+    // If we output anything, we return failure, which indicates dinit should
+    // terminate before the log is initalised and flushed.
+    using std::cerr;
+    using std::endl;
+
+    char *current_context = nullptr;
+    char *file_context = nullptr;
+    security_class_t security_class;
+    char *new_context = nullptr;
+
+    if (is_selinux_enabled() == 1) {
+        return true;
+    }
+
+    int enforce = 0;
+    // We don't need to worry about the enforcing=0 kernel cmdline option or
```

Comment:
Please extend comments to the correct line width before wrapping them. See CODE-STYLE.